java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -0,0 +1,219 @@
+import org.apache.arrow.vector.BitVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.Float8Vector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.VarBinaryVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.avro.Schema;
+import org.apache.avro.Schema.Type;
+import org.apache.avro.io.Decoder;
+
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+/**
+ * Class that does most of the work to convert Avro data into Arrow columnar format Vector objects.
+ */
+public class AvroToArrowUtils {
+
+  private static final int DEFAULT_BUFFER_SIZE = 256;
+
+  /**
+   * Creates an {@link org.apache.arrow.vector.types.pojo.ArrowType} from the {@link Schema.Field}
+   *
+   <p>This method currently performs following type mapping for Avro data types to corresponding Arrow data types.
+   *
+   * <ul>
+   *   <li>STRING --> ArrowType.Utf8</li>
+   *   <li>INT --> ArrowType.Int(32, signed)</li>
+   *   <li>LONG --> ArrowType.Int(64, signed)</li>
+   *   <li>FLOAT --> ArrowType.FloatingPoint(FloatingPointPrecision.SINGLE)</li>
+   *   <li>DOUBLE --> ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE)</li>
+   *   <li>BOOLEAN --> ArrowType.Bool</li>
+   *   <li>BYTES --> ArrowType.Binary</li>
+   * </ul>
+   */
+  private static ArrowType getArrowType(Type type) {
+
+    Preconditions.checkNotNull(type, "Avro type object can't be null");
+
+    final ArrowType arrowType;
+
+    switch (type) {
+      case STRING:
+        arrowType = new ArrowType.Utf8();
+        break;
+      case INT:
+        arrowType = new ArrowType.Int(32, /*signed=*/true);
+        break;
+      case BOOLEAN:
+        arrowType = new ArrowType.Bool();
+        break;
+      case LONG:
+        arrowType = new ArrowType.Int(64, /*signed=*/true);
+        break;
+      case FLOAT:
+        arrowType = new ArrowType.FloatingPoint(SINGLE);
+        break;
+      case DOUBLE:
+        arrowType = new ArrowType.FloatingPoint(DOUBLE);
+        break;
+      case BYTES:
+        arrowType = new ArrowType.Binary();
+        break;
+      default:
+        // no-op, shouldn't get here
+        throw new RuntimeException("Can't convert avro type %s to arrow type." + type.getName());
+    }
+
+    return arrowType;
+  }
+
+  /**
+   * Create Arrow {@link org.apache.arrow.vector.types.pojo.Schema} object for the given Avro {@link Schema}.
+   */
+  public static org.apache.arrow.vector.types.pojo.Schema avroToArrowSchema(Schema schema) {
+
+    Preconditions.checkNotNull(schema, "Avro Schema object can't be null");
+    List<Field> arrowFields = new ArrayList<>();
```

Comment:
nit: initialize with size?

Reply:
Size is different with different avro type.

Reply:
Size is different with different avro type (e.g. primitive type size=1 and record type size > 1), if we initialize this with Field[], its not convenient to process later.